wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -891,27 +891,24 @@ public final class SessionManager: NSObject, SessionManagerType {
         }
     }
 
-    private func switchAnalyticsUser() {
+    private func switchAnalyticsUser(to userSession: ZMUserSession) {
         guard
             let analyticsManager,
-            let activeUserSession,
-            let userProfile = getUserAnalyticsProfileForActiveUserSession()
+            let userProfile = getUserAnalyticsProfile(for: userSession)
         else {
             return
         }
 
         let analyticsSession = analyticsManager.switchUser(userProfile)
-
-        activeUserSession.analyticsSession = analyticsSession
+        userSession.analyticsSession = analyticsSession
     }
 
-    func getUserAnalyticsProfileForActiveUserSession() -> AnalyticsUserProfile? {
-        guard let activeUserSession else { return nil }
-        let selfUser = ZMUser.selfUser(inUserSession: activeUserSession)
+    func getUserAnalyticsProfile(for userSession: ZMUserSession) -> AnalyticsUserProfile? {
+        let selfUser = ZMUser.selfUser(inUserSession: userSession)
 
         // Set the analytics identifier if it's not present
         if selfUser.analyticsIdentifier == nil {
-            let idProvider = AnalyticsIdentifierProvider(selfUser: activeUserSession.selfUser)
+            let idProvider = AnalyticsIdentifierProvider(selfUser: userSession.selfUser)
```

Comment:
It's confusing that you access the self user it two different ways. I suggest to reuse the self user stored at the beginning of the method.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/edeedc9c3a03de80d18e046fbe7705f22943fe27